sympy/ntheory/ecm.py

```diff
@@ -0,0 +1,309 @@
+    the fact that even if kP != 0, the value of k might miss just one large
+    prime divisor of |E(FF(q))|. In this case we only need to compute the
+    scalar multiplication by p to get p*k*P = O. Here a second bound B2
+    restrict the size of possible values of p.
+
+    Parameters:
+    ===========
+
+    n : Number to be Factored
+    B1 : Stage 1 Bound
+    B2 : Stage 2 Bound
+    max_curve : Maximum number of curves generated
+
+    References
+    ==========
+
+    .. [1]  Carl Pomerance and Richard Crandall "Prime Numbers:
+        A Computational Perspective" (2nd Ed.), page 344
+    """
+    from sympy import gcd, mod_inverse, sqrt
+    n = as_int(n)
+    if B1 % 2 != 0 or B2 % 2 != 0:
+        raise ValueError("The Bounds should be an even integer")
+    sieve.extend(B2)
+
+    if isprime(n):
+        return n
+
+    curve = 0
+    D = int(sqrt(B2))
+    beta = [0]*(D + 1)
+    S = [0]*(D + 1)
+    k = 1
+    for p in sieve.primerange(1, B1 + 1):
+        k *= pow(p, integer_log(B1, p)[0])
+    g = 1
+
+    while(curve <= max_curve):
+        curve += 1
+
+        #Suyama's Paramatrization
+        sigma = random.randint(6, n - 1)
+        u = (sigma*sigma - 5) % n
+        v = (4*sigma) % n
+        diff = v - u
+        u_3 = pow(u, 3, n)
+
+        try:
+            C = (pow(diff, 3, n)*(3*u + v)*mod_inverse(4*u_3*v, n) - 2) % n
+        except ValueError:
+            #If the mod_inverse(4*u_3*v, n) doesn't exist
+            return gcd(4*u_3*v, n)
+
+        a24 = (C + 2)*mod_inverse(4, n) % n
+        Q = Point(u_3 , pow(v, 3, n), a24, n)
+        Q = Q.mont_ladder(k)
+        g = gcd(Q.z_cord, n)
+
+        #Stage 1 factor
+        if g != 1 and g != n:
+            return g
+        #Stage 1 failure. Q.z = 0, Try another curve
+        elif g == n:
+            continue
+
+        #Stage 2 - Improved Standard Continuation
+        S[1] = Q.double()
+        S[2] = S[1].double()
+        beta[1] = (S[1].x_cord*S[1].z_cord) % n
+        beta[2] = (S[2].x_cord*S[2].z_cord) % n
+
+        for d in range(3, D + 1):
+            S[d] = S[d - 1].add(S[1], S[d - 2])
+            beta[d] = (S[d].x_cord*S[d].z_cord) % n
+
+        g = 1
+        B = B1 - 1
+        T = Q.mont_ladder(B - 2*D)
+        R = Q.mont_ladder(B)
+
+        for r in  range(B, B2, 2*D):
+            alpha = (R.x_cord*R.z_cord) % n
+            for q in sieve.primerange(r + 2, r + 2*D + 1):
+                delta = (q - r) // 2
+                f = (R.x_cord - S[d].x_cord)*(R.z_cord + S[d].z_cord) -\
+                alpha + beta[delta]
+                g = (g*f) % n
+            #Swap
+            TT = R
+            R = R.add(S[D], T)
+            T = TT
+        g = gcd(n, g)
+
+        #Stage 2 Factor found
+        if g != 1 and g != n:
+            return g
+
+    #ECM failed, Increase the bounds
+    raise ValueError("Increase the bounds")
+
+
+def ecm(n, B1=10000, B2=100000, max_curve=200, seed=1234):
```

Comment:
It could have default value `None` in which case the code would not set the seed. But it could be useful to have in case someone would want to have a repeatable procedure for optimizing the parameters, for example. This function is not intended for ordinary users. Only `factorint` would invoke `ecm` under certain conditions.